domain/blockdag/validate.go

```diff
@@ -936,10 +936,9 @@ func CheckTransactionInputsAndCalulateFee(
 		return 0, nil
 	}
 
-	txID := tx.ID()
 	var totalSompiIn uint64
 	for txInIndex, txIn := range tx.MsgTx().TxIn {
-		entry, err := checkReferencedOutputsAreAvailable(tx, utxoSet, txIn, txInIndex)
+		entry, err := findReferencedOutput(tx, utxoSet, txIn, txInIndex)
```

Comment:
I still don't like the fact that the only use of findReferencedOutput is to wrap the error behind utxoSet.Get(). Don't you prefer to just inline it?

Reply:
The same is more or less true for validateCoinbaseMaturity, checkEntryAmounts, etc.
But this makes CheckTransactionInputsAndCalulateFee much more readable, where you can with a single glance understand all that it does.